src/main/java/seedu/address/storage/loanrecords/JsonAdaptedLoan.java

```diff
@@ -0,0 +1,96 @@
+package seedu.address.storage.loanrecords;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.book.SerialNumber;
+import seedu.address.model.borrower.BorrowerId;
+import seedu.address.model.loan.Loan;
+
+/**
+ * Jackson-friendly version of {@link Loan}.
+ */
+class JsonAdaptedLoan {
+
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Loan's %s field is missing!";
+    public static final String DATE_MESSAGE_CONSTRAINTS = "Date should be in the ISO-8601 format uuuu-MM-dd";
```

Comment:
Should we change "uuuu" to "yyyy" for better readability (and understandability)?